department-of-veterans-affairs/abd-vro
# MCP-2043: add tests for MAS routes - possible paths

## What was the problem?

MAS integration follows a complex workflow combining several external services. Calls to services can result in a fork of the logic path or in an exception. We need to ensure that the software correctly covers the happy paths and can also handle exceptional conditions.

Associated tickets or Slack threads:

- [MCP-2043](https://amida.atlassian.net/browse/MCP-2043)

## How does this fix it?[^1]

This ticket adds positive and negative test cases for Mas integration and adds some more error handling. Whereas the test scenarios do not cover every single case, the framework is established so we can build upon it and make sure future code changes do not introduce regressions.

## How to test this PR
- Execute "automatedClaim" with a collection that has sufficient evidence. 
- Execute "automatedClaim" with a collection that has insufficient evidence. At the moment "orderExam" is not implemented so it will throw an exception

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
